lib/src/simulations/epicycloid_pedal.dart

```diff
@@ -0,0 +1,368 @@
+import 'dart:math';
+import 'dart:ui';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+GlobalKey<_EpicycloidPedalState> globalKey = GlobalKey<_EpicycloidPedalState>();
+
+List<Offset> ys = [];
+
+class EpicycloidPedalCurve extends StatefulWidget {
+  @override
+  _EpicycloidPedalCurveState createState() => _EpicycloidPedalCurveState();
+}
+
+class _EpicycloidPedalCurveState extends State<EpicycloidPedalCurve> {
+  double time = 0;
+  double amplitudeRadius = 35;
+  double _k = 1;
+  double f = 0.01;
+  bool animate = false;
+  bool animating = false;
+
+  @override
+  void initState() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+    ]);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    ys.clear();
+    time = 0;
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+      DeviceOrientation.landscapeLeft,
+      DeviceOrientation.landscapeRight,
+    ]);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    ScreenUtil.instance = ScreenUtil(
+      width: MediaQuery.of(context).size.width,
+      height: MediaQuery.of(context).size.height,
```

Comment:
This has to be constant values of the designers mobile device, you can print them in the terminal and change them to that.